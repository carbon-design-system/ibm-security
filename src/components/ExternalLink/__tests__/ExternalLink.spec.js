/**
 * @file External link tests.
 * @copyright IBM Security 2020
 */

import { render } from '@testing-library/react';
import React from 'react';

import { ExternalLink } from '../../../..';

describe('ExternalLink', () => {
  test('should have no Axe or DAP violations', async () => {
    const main = document.createElement('main');
    render(
      <ExternalLink href="https://www.ibm.com/security">
        test link
      </ExternalLink>,
      {
        // DAP requires a landmark '<main>' in the DOM:
        container: document.body.appendChild(main),
      }
    );
    await expect(document.body).toHaveNoAxeViolations();
    await expect(document.body).toHaveNoDAPViolations('ExternalLink');
  });

  test('should add children as link text', () => {
    const { queryByText } = render(
      <ExternalLink href="https://www.ibm.com/security">test link</ExternalLink>
    );
    expect(queryByText(/test link/i)).toBeVisible();
  });

  test('should add an href', () => {
    const { getByText } = render(
      <ExternalLink href="https://www.ibm.com/security">test link</ExternalLink>
    );
    expect(getByText(/test link/i).closest('a')).toHaveAttribute(
      'href',
      'https://www.ibm.com/security'
    );
  });

  test('should add custom class', () => {
    const { getByText } = render(
      <ExternalLink
        href="https://www.ibm.com/security"
        className="custom-class"
      >
        test link
      </ExternalLink>
    );
    expect(getByText(/test link/i).closest('a')).toHaveClass('custom-class');
  });

  test('should pass through extra props via spread attribute', () => {
    const { queryByTestId } = render(
      <ExternalLink href="https://www.ibm.com/security" data-testid="test-id">
        test link
      </ExternalLink>
    );
    expect(queryByTestId('test-id')).toBeInTheDocument();
  });
});
