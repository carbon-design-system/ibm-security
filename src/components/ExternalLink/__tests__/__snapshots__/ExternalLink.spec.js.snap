// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink Rendering renders correctly 1`] = `
<Link
  className="security--link--external"
  href="https://www.ibm.com/security"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    className="security--link--external__title"
  >
    https://www.ibm.com/security
  </span>
  <span
    className="security--link--external__icon-wrapper"
  >
    <Icon
      className="security--link--external__icon"
      path={null}
      renderIcon={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      size={20}
      viewBox="0 0 32 32"
    />
  </span>
</Link>
`;

exports[`ExternalLink Rendering renders the HTML of the node's subtree 1`] = `
<a
  class="bx--link security--link--external"
  href="https://www.ibm.com/security"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    class="security--link--external__title"
  >
    https://www.ibm.com/security
  </span>
  <span
    class="security--link--external__icon-wrapper"
  >
    <svg
      aria-hidden="true"
      class="security--icon security--link--external__icon"
      focusable="false"
      height="16"
      preserveAspectRatio="xMidYMid meet"
      style="will-change:transform"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13 14H3c-.6 0-1-.4-1-1V3c0-.6.4-1 1-1h5v1H3v10h10V8h1v5c0 .6-.4 1-1 1z"
      />
      <path
        d="M10 1v1h3.3L9 6.3l.7.7L14 2.7V6h1V1z"
      />
    </svg>
  </span>
</a>
`;
